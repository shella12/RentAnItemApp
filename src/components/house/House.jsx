import React from 'react';
import PropTypes from 'prop-types';
import './house.css';

const House = (props) => {
  const { data, handleRemove } = props;
  const {
    id, name, price, description,
  } = data;

  return (
    <>
      <div className="column house">
        <img src={data.picture_url} alt={`house ${name}`} />
        <div className="column house-content-wrapper">
          <div className="column max-content-flex house-content">
            <div className="row space-between">
              <h4 className="max-content-flex house-title">{name}</h4>
              <h4>
                $
                {price}
                {' '}
                <br />
                per Month
              </h4>
            </div>
            <p className="desktop-only house-description">{description}</p>
          </div>
          {
            handleRemove
            && (
            <button
              type="button"
              className="add-favorite btn-default"
              onClick={() => handleRemove(id)}
            >
              Delete
            </button>
            )
          }
        </div>
      </div>
    </>
  );
};

House.defaultProps = {
  handleRemove: () => {},
};

House.propTypes = {
  data: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    description: PropTypes.string,
    price: PropTypes.string,
    picture_url: PropTypes.string,
  }).isRequired,
  handleRemove: PropTypes.func,
};

export default House;
