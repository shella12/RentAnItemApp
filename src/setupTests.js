import '@testing-library/jest-dom';
import fetchMock from 'fetch-mock';

export const testListHouses = [{
  id: 1, name: 'Pent house', price: '300000.0', picture: 'https://github.com/shella12/RentAnItemBackend/assets/44798044/5bb4ea2c-4ae5-45b8-9083-aeafbfbcf287', description: "This house has 5 rooms upstairs and 4 rooms downstairs. Home design emphasizes clean lines and geometric shapes.\n         The magic thing about this houe is that it feels good to leave, and it feels even better to come back.\n         Modern Conveniences and Green Features. Well-Placed Bathrooms. Quality Construction with Distinct Architecture. \n         Plenty of Storage Space. It's important to have designated storage areas and large closets.", owner_name: 'susan perry', created_at: '2023-05-12T12:38:59.989Z', updated_at: '2023-05-12T12:38:59.989Z',
}, {
  id: 6, name: 'Mansion', price: '870000.0', picture: 'https://github.com/shella12/RentAnItemBackend/assets/44798044/ac143109-1ae3-4885-a650-808b3bdb6cda', description: "This house has 5 rooms upstairs and 4 rooms downstairs. Home design emphasizes clean lines and geometric shapes.\n         The magic thing about this houe is that it feels good to leave, and it feels even better to come back.\n         Modern Conveniences and Green Features. Well-Placed Bathrooms. Quality Construction with Distinct Architecture. \n         Plenty of Storage Space. It's important to have designated storage areas and large closets.", owner_name: 'Kim buffey', created_at: '2023-05-12T12:39:00.010Z', updated_at: '2023-05-12T12:39:00.010Z',
}];

export const testListFavoriteHouses = [
  {
    id: 1,
    name: 'Pent house',
    price: '300000.0',
    picture: 'https://github.com/shella12/RentAnItemBackend/assets/44798044/5bb4ea2c-4ae5-45b8-9083-aeafbfbcf287',
    description: "This house has 5 rooms upstairs and 4 rooms downstairs. Home design emphasizes clean lines and geometric shapes.\n         The magic thing about this houe is that it feels good to leave, and it feels even better to come back.\n         Modern Conveniences and Green Features. Well-Placed Bathrooms. Quality Construction with Distinct Architecture. \n         Plenty of Storage Space. It's important to have designated storage areas and large closets.",
    owner_name: 'susan perry',
    created_at: '2023-05-12T12:38:59.989Z',
    updated_at: '2023-05-12T12:38:59.989Z',
  },
];

export const fetchFavoriteHouseData = (userID) => {
  fetchMock.getOnce(`https://renthousebackend.onrender.com/api/v1/users/${userID}/favorite_houses`, {
    headers: { 'content-type': 'application/json' },
    body: testListFavoriteHouses,
  });
};

export const fetchHouseData = () => {
  fetchMock.getOnce('https://renthousebackend.onrender.com/api/v1/houses', {
    headers: { 'content-type': 'application/json' },
    body: testListHouses,
  });
};
