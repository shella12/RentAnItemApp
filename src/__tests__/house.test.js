import renderer, { act } from 'react-test-renderer';
import '@testing-library/jest-dom';
import React from 'react';
import { waitFor } from '@testing-library/react';
import { testListHouses } from '../setupTests';
import House from '../components/house/House';

describe('test render', () => {
  test('House should match snapshoot', async () => {
    let tree;
    act(() => {
      tree = renderer.create(
        <React.StrictMode>
          <House data={testListHouses[0]} />
        </React.StrictMode>,
      );
    });

    await waitFor(() => {
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
