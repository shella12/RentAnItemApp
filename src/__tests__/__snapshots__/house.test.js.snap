// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render House should match snapshoot 1`] = `
<div
  className="column house"
>
  <img
    alt="house Pent house"
  />
  <div
    className="column house-content-wrapper"
  >
    <div
      className="column max-content-flex house-content"
    >
      <div
        className="row space-between"
      >
        <h4
          className="max-content-flex house-title"
        >
          Pent house
        </h4>
        <h4>
          $
          300000.0
           
          <br />
          per Month
        </h4>
      </div>
      <p
        className="desktop-only house-description"
      >
        This house has 5 rooms upstairs and 4 rooms downstairs. Home design emphasizes clean lines and geometric shapes.
         The magic thing about this houe is that it feels good to leave, and it feels even better to come back.
         Modern Conveniences and Green Features. Well-Placed Bathrooms. Quality Construction with Distinct Architecture. 
         Plenty of Storage Space. It's important to have designated storage areas and large closets.
      </p>
    </div>
    <button
      className="add-favorite btn-default"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </div>
</div>
`;
